services/consensusalgo/benchmarkconsensus/test/non_leader_test.go

```diff
@@ -2,39 +2,135 @@ package test
 
 import (
 	"context"
-	"github.com/orbs-network/go-mock"
 	"github.com/orbs-network/orbs-network-go/test"
 	"github.com/orbs-network/orbs-network-go/test/builders"
-	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
-	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
 	"testing"
 )
 
 func TestNonLeaderDoesNotCreateBlocks(t *testing.T) {
 	test.WithContext(func(ctx context.Context) {
 		h := newHarness(false)
-		h.consensusContext.Reset().When("RequestNewTransactionsBlock", mock.Any).Return(nil, nil).Times(0)
+		h.expectNoBlockCreation()
 		h.createService(ctx)
-		err := test.ConsistentlyVerify(h.consensusContext)
-		if err != nil {
-			t.Fatal("Did create block with ConsensusContext:", err)
-		}
+		h.verifyNoBlockCreation(t)
 	})
 }
 
-func TestNonLeaderIgnoreFutureBlockHeight(t *testing.T) {
+func TestNonLeaderCommitsAndRepliesToValidBlockHeights(t *testing.T) {
 	test.WithContext(func(ctx context.Context) {
 		h := newHarness(false)
 		h.createService(ctx)
-		h.gossip.Reset().When("SendBenchmarkConsensusCommitted", mock.Any).Return(nil, nil).Times(0)
-		h.service.HandleBenchmarkConsensusCommit(&gossiptopics.BenchmarkConsensusCommitInput{
-			Message: &gossipmessages.BenchmarkConsensusCommitMessage{
-				BlockPair: builders.BlockPair().WithHeight(1000).Build(),
-			},
-		})
-		err := test.ConsistentlyVerify(h.gossip)
-		if err != nil {
-			t.Fatal("Did not ignore block with future block height:", err)
+
+		var replyBlockHeight primitives.BlockHeight
+
+		h.expectCommitSaveAndReply(&replyBlockHeight)
+		b1 := builders.BlockPair().
+			WithHeight(1).
+			WithBenchmarkConsensusBlockProof(nil, h.config.ConstantConsensusLeader()). //TODO: fix private key
+			Build()
+		h.receiveCommit(b1)
+		if replyBlockHeight != 1 {
+			t.Fatalf("Replied committed with wrong last block height: %d", replyBlockHeight)
+		}
+		h.verifyCommitSaveAndReply(t)
+
+		h.expectCommitSaveAndReply(&replyBlockHeight)
+		b2 := builders.BlockPair().
```

Comment:
maybe the builder can have a `ThatIsNextOf(prevBlock)` method, that automatically sets height and prev block hash

Reply:
It does, it's called `WithPrevBlockHash(prevBlock *protocol.BlockPairContainer)`

it takes a previous block and sets the hashes accordingly